### PR TITLE
Makefile: don't allocate pseudo-tty if not interactive

### PR DESCRIPTION
**Pull request by jlebon**
In commit 00030ce, the DOCKER_RUN_DOCKER variable was introduced to
simplify the Makefile. However, as a result, because the `shell` target
required `-ti`, the DOCKER_RUN_DOCKER variable was made to have `-ti` as
well.

In this commit, we remove the default pseudo-tty allocation for targets
that don't require it (all of them except the `shell` target). This is
useful in testing contexts where we want to redirect/pipe the output
from the tests.

We also remove the DOCKER_RUN_DOCS variable, which is no longer in use
in this Makefile (seems like the related targets that used it were
migrated to docs/Makefile).

Signed-off-by: Jonathan Lebon jlebon@redhat.com

### CODE REVIEW

**Comment by cpuguy83** [comment]
ping @tianon @jfrazelle

**Comment by jfrazelle** [comment]
We need -i for tests

**Comment by tianon** [comment]
Don't we need "-t" for the tests too, since some of them need to allocate
additional ttys inside the container?  Or does "--privileged" already give
us what we need for that to work?

**Comment by jfrazelle** [comment]
We don't add the -t on Jenkins 






On Wed, Oct 21, 2015 at 9:23 AM -0700, "Tianon Gravi" <notifications@github.com> wrote:










Don't we need "-t" for the tests too, since some of them need to allocate

additional ttys inside the container?  Or does "--privileged" already give

us what we need for that to work?

**Comment by tianon** [comment]
Ah, then privileged must be enough for that. :smile:

**Comment by jfrazelle** [comment]
must be :) @jlebon can you update the pr

**Comment by jlebon** [comment]
@jfrazelle updated, thanks for having a look.

**Comment by jlebon** [comment]
Do you have some tests other than the CI ones that depend on `-i`? Seems like they all passed with the original patch removing `-ti`. But either way, this works just as well for me. :)

Also, you mentioned

> We don't add the -t on Jenkins 

I'm interested in this since I'm actually working on Jenkins integration as well for [projectatomic/docker](https://github.com/projectatomic/docker). In what other ways do you modify the test suite for Jenkins? (Or is there a document that already details this?)

**Comment by jfrazelle** [comment]
we dont modify anything except that

**Comment by jfrazelle** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -69,7 +67,7 @@ validate: build
 	$(DOCKER_RUN_DOCKER) hack/make.sh validate-dco validate-gofmt validate-pkg validate-lint validate-test validate-toml validate-vet
 
 shell: build
-	$(DOCKER_RUN_DOCKER) bash
+	$(DOCKER_RUN_DOCKER) -t bash
```
Comment (jfrazelle) [inline Makefile]: this will not work, the image was before the flag

**Comment by jlebon** [comment]
@jfrazelle I realize now that we actually had the same problem which caused you to have to remove the `-t` for Jenkins. It was not piping exactly that was causing it to fail but the fact that the session was not interactive. The patch set now checks for that and adds `-t` only if the session is interactive, which should allow you to avoid having to manually disable `-t` in Jenkins jobs.

**Comment by jlebon** [comment]
@jfrazelle Could you have a look at the latest commit?

**Comment by jfrazelle** [comment]
LGTM

**Comment by jlebon** [comment]
Ping @tianon, could you take a look when you can?

**Comment by tianon** [comment]
LGTM

**Comment by thaJeztah** [comment]
@tianon erm, merge?

**Comment by thaJeztah** [comment]
@runcom gave me his "lgtm" as well, YOLO: merging \o/

**Pull request merged**